elmslie-coroutines/src/main/java/vivid/money/elmslie/core/ElmStoreCompat.kt

```diff
@@ -0,0 +1,59 @@
+package vivid.money.elmslie.core
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.launch
+import vivid.money.elmslie.core.disposable.Disposable
+import vivid.money.elmslie.core.store.DefaultActor
+import vivid.money.elmslie.core.store.StateReducer
+import vivid.money.elmslie.core.store.ElmStore
+import vivid.money.elmslie.core.store.Store
+
+/**
+ * Compatibility [Store] implementation applying coroutines for multithreading.
+ */
+class ElmStoreCompat<Event : Any, State : Any, Effect : Any, Command : Any>(
+    initialState: State,
+    reducer: StateReducer<Event, State, Effect, Command>,
+    actor: Actor<Command, out Event>
+) : Store<Event, Effect, State> by ElmStore(
+    initialState = initialState,
+    reducer = reducer,
+    actor = actor.toActor()
+)
+
+@Suppress("TooGenericExceptionCaught")
+private fun <Command : Any, Event : Any> Actor<Command, Event>.toActor() =
+    DefaultActor<Command, Event> { command, onEvent, onError ->
+        val job = GlobalScope.launch(Dispatchers.Unconfined) {
+            try {
+                execute(command).flowOn(Dispatchers.IO).collect { event -> onEvent(event) }
+            } catch (t: Throwable) {
+                onError(t)
```

Comment:
As I remember we shouldn't catch the generic exceptions in coroutines, because of CancellationException. Let's add check to be sure that CancellationException is correctly propagated.
```
        if (e is CancellationException) {
            throw e
        }
```

Reply:
Thank you, i've updated the sources